pkg/cluster/deploystorage.go

```diff
@@ -191,6 +186,144 @@ func (m *manager) deployStorageTemplate(ctx context.Context, installConfig *inst
+					"Microsoft.Compute/virtualMachineScaleSets/virtualMachines/extensions/delete",
+					"Microsoft.Compute/virtualMachineScaleSets/delete/action",
+					"Microsoft.Compute/virtualMachineScaleSets/start/action",
+					"Microsoft.Compute/virtualMachineScaleSets/redeploy/action",
+					"Microsoft.Compute/virtualMachineScaleSets/scale/action",
+					"Microsoft.Compute/virtualMachineScaleSets/extensions/write",
+					"Microsoft.Compute/virtualMachineScaleSets/extensions/delete",
+
+					"Microsoft.Network/dnszones/A/write",
+					"Microsoft.Network/dnszones/A/delete",
+					"Microsoft.Network/dnszones/AAAA/write",
+					"Microsoft.Network/dnszones/AAAA/delete",
+					"Microsoft.Network/dnszones/CNAME/write",
+					"Microsoft.Network/dnszones/CNAME/delete",
+
+					"Microsoft.Network/privateDnsZones/A/write",
+					"Microsoft.Network/privateDnsZones/A/delete",
+					"Microsoft.Network/privateDnsZones/AAAA/write",
+					"Microsoft.Network/privateDnsZones/AAAA/delete",
+					"Microsoft.Network/privateDnsZones/CNAME/write",
+					"Microsoft.Network/privateDnsZones/CNAME/delete",
+
+					"Microsoft.ClassicStorage/storageAccounts/write",
+					"Microsoft.ClassicStorage/storageAccounts/delete",
+					"Microsoft.ClassicStorage/storageAccounts/listKeys/action",
+					"Microsoft.ClassicStorage/storageAccounts/disks/write",
+					"Microsoft.ClassicStorage/storageAccounts/disks/delete",
+
+					"Microsoft.Storage/storageAccounts/updateInternalProperties/action",
+					"Microsoft.Storage/storageAccounts/listkeys/action",
+					"Microsoft.Storage/storageAccounts/delete",
+					"Microsoft.Storage/storageAccounts/write",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/write",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/lease/action",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/write",
+				},
+				&[]string{},
+				&[]string{
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/deleteBlobVersion/action",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/permanentDelete/action",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/add/action",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/filter/action",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/move/action",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/tags/read",
+					"Microsoft.Storage/storageAccounts/blobServices/containers/blobs/tags/write",
+				},
+				&[]string{}),
+			rbac.ResourceGroupCustomRoleAssignmentWithName(
+				"5a33ff5e-e2be-516a-900c-195f3e6e0c4e",
+				//"guid(resourceGroup().id, 'SP / RG CSP')",
```

Comment:
needs fixing up

Reply:
fixed, and accepts custom role name as the parameter.